publisher-sdk-tests/src/androidTest/java/com/criteo/publisher/integration/ProfileIdFunctionalTest.kt

```diff
@@ -53,8 +56,12 @@ class ProfileIdFunctionalTest {
   @SpyBean
   private lateinit var api: PubSdkApi
 
+  @MockBean
```

Comment:
If you use `SpyBean` you could remove all the stubs, except at line 291.
With it, the rate limiter will act normally, and as the shared pref are cleaned between tests, any tests can run a prefetch if it wants to. In case of the line 291, the memory is cleaned, but the persisted data is not in purpose, hence you'll need the stub there.